dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -2526,6 +2529,89 @@ BlockInputStreamPtr Segment::getBitmapFilterInputStream(BitmapFilterPtr && bitma
         dm_context.tracing_id);
 }
 
+BlockInputStreamPtr Segment::getLateMaterializationStream(BitmapFilterPtr && bitmap_filter,
+                                                          const DMContext & dm_context,
+                                                          const ColumnDefines & columns_to_read,
+                                                          const SegmentSnapshotPtr & segment_snap,
+                                                          const RowKeyRanges & data_ranges,
+                                                          const PushDownFilterPtr & filter,
+                                                          UInt64 max_version,
+                                                          size_t expected_block_size)
+{
+    auto enable_handle_clean_read = !hasColumn(columns_to_read, EXTRA_HANDLE_COLUMN_ID);
+    constexpr auto is_fast_scan = true;
+    auto enable_del_clean_read = !hasColumn(columns_to_read, TAG_COLUMN_ID);
+
+    const auto & filter_columns = filter->filter_columns;
+    SkippableBlockInputStreamPtr filter_column_stable_stream = segment_snap->stable->getInputStream(
+        dm_context,
+        filter_columns,
+        data_ranges,
+        filter->rs_operator,
+        max_version,
+        expected_block_size,
+        enable_handle_clean_read,
+        is_fast_scan,
+        enable_del_clean_read);
+
+    auto filter_columns_to_read_ptr = std::make_shared<ColumnDefines>(filter_columns);
+    SkippableBlockInputStreamPtr filter_column_delta_stream = std::make_shared<DeltaValueInputStream>(
+        dm_context,
+        segment_snap->delta,
+        filter_columns_to_read_ptr,
+        this->rowkey_range);
+
+    BlockInputStreamPtr filter_column_stream = std::make_shared<RowKeyOrderedBlockInputStream>(
```

Comment:
It's better to use ConcatSkippableBlockInputStream instead of RowKeyOrderedBlockInputStream, and we may don't need RowKeyOrderedBlockInputStream. 

Besides, the name of RowKeyOrderedBlockInputStream is kind of weird and inappropriate.

Reply:
Keep it temporarily....